### PR TITLE
Rudy/testing

### PR DESCRIPTION
**Pull request by Rodolfoguzman25**
Changed 'joesFrontEndCookieParser' to cookieParser.
NPM installed jest, sinon, enzyme, and supertest.
Created automated test for get request to yelp API, and cookieParser.

### CODE REVIEW

File: testing/authenticate.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const authenticate = require('../client/services/authenticate');
+
+test('authenticate.cookieParser' , ()=>{
+    expect(authenticate.cookieParser('bobby= henry; sally=harry; christian= freddy;'))
```
Comment (conor-jh) [inline testing/authenticate.test.js]: @Rodolfoguzman25 Can you add `describe` and/or `it` calls here to clarify what’s being tested? With the hard-coded data, it’s a little unclear… 🙏

File: testing/server.test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+const request = require('supertest');
+const server = 'http://localhost:3000'
+const app = require('../server/server');
+
+
+describe('verifyUser', () => {
+  describe('/', () => {
+    describe('/login', () => {
+      it('responds with 200 status and text/html content type', () => {
```
Comment (conor-jh) [inline testing/server.test.js]: Change `it` to expect `application/json` content type to reflect the actual test

File: testing/server.test.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          .expect('Content-Type', /application\/json/)
+          .expect(200);      
+      })
+    });
+  });
+});
+
+describe('searchYelp', () => {
+    describe('GET', () => {
+      it('responds with an empty json object if name/zip are undefined', () => {
```
Comment (conor-jh) [inline testing/server.test.js]: Update this test description to match the test (responds with 404… not clear what the response body is from the test itself).

Should also probably test expect assertion for missing name **and** missing zip

File: testing/server.test.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+      it('responds with an object containing all restaurants and locations', () => {
+        return request(app)
+          .get('/yelp/restaurantName/chipotle/restaurantZip/10021')
+          .expect('Content-Type', 'application/json; charset=utf-8')
+          .expect(200)
+          .then(data => {
+            expect(data.length > 1);
+          });
+      });
```
Comment (conor-jh) [inline testing/server.test.js:41]: Doesn’t have to be part of this pull request, but it’d probably be good to add tests to check the proper data points are coming back… e.g. 
```js
.expect(data[0].name).toBe('Chipotle')
```

**Review by conor-jh — CHANGES_REQUESTED** [review]
Hey Rudy, these look good! Check out my inline comments in some of your test files… just a few things to clarify before we merge in 👍

**Review by conor-jh — APPROVED** [review]
Thanks for those changes, looks good 🃏

**Review by Donaldblanc — APPROVED** [review]
nice work on the testing using Regex is a clever way to ensure the correct data is received

**Comment by Donaldblanc** [comment]
Awesome Job man, keep up the good work.

**Pull request merged**